packages/databricks-vscode/src/test/e2e/bundle_variables.e2e.ts

```diff
@@ -0,0 +1,217 @@
+import assert from "node:assert";
+import {
+    dismissNotifications,
+    getUniqueResourceName,
+    getViewSection,
+    waitForLogin,
+    waitForTreeItems,
+} from "./utils/commonUtils.ts";
+import {CustomTreeSection, TextEditor, Workbench} from "wdio-vscode-service";
+import {
+    getBasicBundleConfig,
+    writeRootBundleConfig,
+} from "./utils/dabsFixtures.ts";
+import {BundleSchema} from "../../bundle/types.ts";
+
+describe("Bundle Variables", async function () {
+    let workbench: Workbench;
+    let vscodeWorkspaceRoot: string;
+    let clusterId: string;
+    let schemaDef: BundleSchema;
+
+    this.timeout(3 * 60 * 1000);
+
+    async function createProjectWithJob() {
+        const projectName = getUniqueResourceName("bundle_variables");
+        schemaDef = getBasicBundleConfig({
+            bundle: {
+                name: projectName,
+                deployment: {},
+            },
+            variables: {
+                varWithDefault: {
+                    default: "default",
+                },
+            },
+            targets: {
+                dev_test: {
+                    variables: {
+                        varWithDefault: "dev",
+                    },
+                },
+            },
+        });
+
+        /* eslint-enable @typescript-eslint/naming-convention */
+        await writeRootBundleConfig(schemaDef, vscodeWorkspaceRoot);
+    }
+
+    before(async function () {
+        assert(
+            process.env.TEST_DEFAULT_CLUSTER_ID,
+            "TEST_DEFAULT_CLUSTER_ID env var doesn't exist"
+        );
+        assert(
+            process.env.WORKSPACE_PATH,
+            "WORKSPACE_PATH env var doesn't exist"
+        );
+        assert(
+            process.env.DATABRICKS_HOST,
+            "DATABRICKS_HOST env var doesn't exist"
+        );
+
+        clusterId = process.env.TEST_DEFAULT_CLUSTER_ID;
+        workbench = await browser.getWorkbench();
+        vscodeWorkspaceRoot = process.env.WORKSPACE_PATH;
+        await createProjectWithJob();
+        await dismissNotifications();
+    });
+
+    it("should wait for extension activation", async () => {
+        const section = await getViewSection("CONFIGURATION");
+        assert(section);
+    });
+
+    it("should wait for connection", async () => {
+        await waitForLogin("DEFAULT");
+        await dismissNotifications();
+    });
+
+    async function assertVariableValue(
+        section: CustomTreeSection,
+        variableName: string,
+        expected: {value?: string; defaultValue?: string; lookup?: string}
+    ) {
+        await section.expand();
+        const variableTreeItem = await section.findItem(variableName);
+        assert(variableTreeItem);
+        console.log(
+            await variableTreeItem.getDescription(),
+            expected.value,
+            (await variableTreeItem.getDescription()) === expected.value
+        );
+        assert((await variableTreeItem.getDescription()) === expected.value);
```

Comment:
Better to use `assert.strictEqual` here (and in similar cases below), instead of simple assert with `===` comparison inside. The `strictEqual` will print all the arguments in case it fails